### PR TITLE
Update to Linter v1.0.0

### LINKED ISSUE #17: Upcoming linter changes
We're in the process of replacing `linter` with a rewrite. Our central issue for tracking the plan is https://github.com/AtomLinter/linter-plus/issues/29. We're working hard to make this a smooth transition for you and for your users.

Please let us know here if you'd prefer to do the work yourself. If you have any questions or concerns please join us in our [room on gitter](https://gitter.im/AtomLinter/Linter).

### LINKED ISSUE #19: Object.activate is deprecated.
AtomLinter v0.X.Y API has been deprecated. Please refer to the Linter docs to update and the latest API: https://github.com/AtomLinter/Linter/wiki/Migrating-to-the-new-API
```
Object.activate (/Users/erik/.atom/packages/linter/lib/main.coffee:57:8)
Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:240:19)
<unknown> (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:221:30)
Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:165:15)
Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:213:14)
PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:452:21)
```

### LINKED ISSUE #20: LinterRuby.Linter is deprecated.
AtomLinter v0.X.Y API has been deprecated. Please refer to the Linter docs to update and the latest API: https://github.com/AtomLinter/Linter/wiki/Migrating-to-the-new-API     

```sh
LinterRuby.Linter - /home/foo/.atom/packages/linter/lib/linter.coffee:60:4
new LinterRuby - /home/foo/.atom/packages/linter-ruby/lib/linter-ruby.coffee:24:4
Object.lint - /home/foo/.atom/packages/linter/lib/legacy.coffee:61:21
<unknown> - /home/foo/.atom/packages/linter/lib/editor-linter.coffee:68:23
<unknown> - /home/foo/.atom/packages/linter/lib/editor-linter.coffee:67:24
Set.forEach - native
```

### PR DESCRIPTION
**Pull request by k2b6s9j**
Fixes #17 
Fixes #19 
Fixes #20

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const regex = "(?<file>.+):(?<line>\\d+):\\s*(?<type>[\\w\\s\\-]+)[:,]?\\s*(?<message>.+)";
+    return {
+      grammarScopes: ["source.ruby", "source.ruby.rails", "source.ruby.rspec"],
+      scope: "file",
+      lintOnFly: true,
+      lint: (activeEditor) => {
+        const command = atom.config.get("linter-ruby.rubyExecutablePath");
+        const file = activeEditor.getPath()
+        const args = ["-wc", file];
+        return helpers.exec(command, args, {stream: "stderr", cwd: file}).then(output =>
```
Comment (steelbrain) [inline lib/main.js]: Instead of defining stuff on line above and using them in only one place, you can inline them.

Comment (Arcanemagus) [inline lib/main.js]: It's nice for readability to have them on separate lines, although _technically_ that is a tad bit of extra memory.

Comment (k2b6s9j) [inline lib/main.js]: I know but I like the shorter line length.

**Comment by k2b6s9j** [comment]
Welp. It's been a day without any major critiques on the code. I'm going to merge and release.

**Comment by steelbrain** [comment]
Maybe because your code was high quality by default @k2b6s9j :)

**Comment by k2b6s9j** [comment]
@steelbrain It's a labor of love.

**Pull request merged**